src/Event/Event.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Event;
+
+// Since PHP-CS-FIXER is PHP 5.6 compliant we can't always use Symfony Contracts (currently needs PHP ^7.1.3)
+// This conditionnal inheritance will be useless when PHP-CS-FIXER no longer supports PHP versions
+// inferior to Symfony/Contracts PHP minimal version
+if (class_exists(\Symfony\Contracts\EventDispatcher\Event::class)) {
+    class BaseEvent extends \Symfony\Contracts\EventDispatcher\Event
+    {
+    }
+} else {
+    class BaseEvent extends \Symfony\Component\EventDispatcher\Event
+    {
+    }
+}
+
+/**
+ * @author Laurent VOULLEMIER <laurent.voullemier@gmail.com>
+ */
+class Event extends BaseEvent
```

Comment:
The `BaseEvent` class looks unnecessary to me. You could directly define the `Event` class within the `if`/`else` block above.

```php
if (class_exists(\Symfony\Contracts\EventDispatcher\Event::class)) {
    class Event extends \Symfony\Contracts\EventDispatcher\Event
    {
    }
} else {
    class Event extends \Symfony\Component\EventDispatcher\Event
    {
    }
}
```

Reply:
Indeed... Thank you